python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -768,63 +771,66 @@ def contains_properly(self, other, align=True):
             A Series of boolean values indicating whether each point falls
             within the corresponding polygon in the input.
         """
-        if not contains_only_polygons(self):
-            raise TypeError(
-                "`.contains` can only be called with polygon series."
-            )
+        return ContainsProperlyBinpred(self, other, align)()
 
-        (lhs, rhs) = self.align(other) if align else (self, other)
-
-        # RHS conditioning:
-        mode = "POINTS"
-        # point in polygon
-        if contains_only_linestrings(rhs):
-            # condition for linestrings
-            mode = "LINESTRINGS"
-            geom = rhs.lines
-        elif contains_only_polygons(rhs) is True:
-            # polygon in polygon
-            mode = "POLYGONS"
-            geom = rhs.polygons
-        elif contains_only_multipoints(rhs) is True:
-            # mpoint in polygon
-            mode = "MULTIPOINTS"
-            geom = rhs.multipoints
-        else:
-            # no conditioning is required
-            geom = rhs.points
-        xy_points = geom.xy
-        point_indices = geom.point_indices()
-        points = GeoSeries(GeoColumn._from_points_xy(xy_points._column)).points
-
-        # call pip on the three subtypes on the right:
-        point_result = contains_properly(
-            points.x,
-            points.y,
-            lhs.polygons.part_offset[:-1],
-            lhs.polygons.ring_offset[:-1],
-            lhs.polygons.x,
-            lhs.polygons.y,
-        )
-        if (
-            mode == "LINESTRINGS"
-            or mode == "POLYGONS"
-            or mode == "MULTIPOINTS"
-        ):
-            # process for completed linestrings, polygons, and multipoints.
-            # Not necessary for points.
-            result = cudf.DataFrame(
-                {"idx": point_indices, "pip": point_result}
-            )
-            # if the number of points in the polygon is equal to the number of
-            # points, then the requirements for `.contains_properly` are met
-            # for this geometry type.
-            df_result = (
-                result.groupby("idx").sum().sort_index()
-                == result.groupby("idx").count().sort_index()
-            )
-            point_result = cudf.Series(
-                df_result["pip"], index=cudf.RangeIndex(0, len(df_result))
-            )
-            point_result.name = None
-        return point_result
+    def intersects(self, other, align=True):
+        """Compute the intersections of two GeoSeries.
```

Comment:
This one just says what it computes, doesn't define intersection.

Reply:
I updated them to have the same short and long text as GeoPandas.